<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="6" passed="2" failed="4" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-02-27T09:56:28 COT" name="ExpressCart Shopping Cart Test Suite" finished-at="2025-02-27T09:57:01 COT" duration-ms="32989">
    <groups>
    </groups>
    <test started-at="2025-02-27T09:56:28 COT" name="Shopping Cart Error Handling Tests" finished-at="2025-02-27T09:56:40 COT" duration-ms="12568">
      <class name="com.expresscart.tests.CartErrorHandlingTest">
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:28 COT" name="setUp" finished-at="2025-02-27T09:56:34 COT" duration-ms="6174" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testInvalidQuantityError()[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:34 COT" name="testInvalidQuantityError" description="Probar manejo de error con cantidad inválida" finished-at="2025-02-27T09:56:37 COT" duration-ms="2367" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInvalidQuantityError -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:37 COT" name="tearDown" finished-at="2025-02-27T09:56:37 COT" duration-ms="665" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testInvalidQuantityError status=SUCCESS method=CartErrorHandlingTest.testInvalidQuantityError()[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:37 COT" name="setUp" finished-at="2025-02-27T09:56:38 COT" duration-ms="1083" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testServerErrorHandling()[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:38 COT" name="testServerErrorHandling" description="Probar manejo de error del servidor (código 500)" finished-at="2025-02-27T09:56:40 COT" duration-ms="1765" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testServerErrorHandling -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3]" started-at="2025-02-27T09:56:40 COT" name="tearDown" finished-at="2025-02-27T09:56:40 COT" duration-ms="207" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testServerErrorHandling status=SUCCESS method=CartErrorHandlingTest.testServerErrorHandling()[pri:0, instance:com.expresscart.tests.CartErrorHandlingTest@7b2bbc3] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- com.expresscart.tests.CartErrorHandlingTest -->
    </test> <!-- Shopping Cart Error Handling Tests -->
    <test started-at="2025-02-27T09:56:28 COT" name="Shopping Cart Functionality Tests" finished-at="2025-02-27T09:57:01 COT" duration-ms="32989">
      <class name="com.expresscart.tests.CartFunctionalityTest">
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:28 COT" name="setUp" finished-at="2025-02-27T09:56:34 COT" duration-ms="6174" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testAddProductToCart()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:34 COT" name="testAddProductToCart" finished-at="2025-02-27T09:56:39 COT" duration-ms="5203" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[El carrito no debería estar vacío expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: El carrito no debería estar vacío expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertFalse(Assert.java:79)
at com.expresscart.tests.CartFunctionalityTest.testAddProductToCart(CartFunctionalityTest.java:39)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testAddProductToCart -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:39 COT" name="tearDown" finished-at="2025-02-27T09:56:40 COT" duration-ms="423" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testAddProductToCart status=FAILURE method=CartFunctionalityTest.testAddProductToCart()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:40 COT" name="setUp" finished-at="2025-02-27T09:56:41 COT" duration-ms="981" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testCartTotalCalculation()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:41 COT" name="testCartTotalCalculation" finished-at="2025-02-27T09:56:44 COT" duration-ms="2696" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[La cantidad del producto debería ser 2 expected [2] but found [0]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: La cantidad del producto debería ser 2 expected [2] but found [0]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.expresscart.tests.CartFunctionalityTest.testCartTotalCalculation(CartFunctionalityTest.java:128)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCartTotalCalculation -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:44 COT" name="tearDown" finished-at="2025-02-27T09:56:44 COT" duration-ms="462" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testCartTotalCalculation status=FAILURE method=CartFunctionalityTest.testCartTotalCalculation()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:44 COT" name="setUp" finished-at="2025-02-27T09:56:45 COT" duration-ms="1197" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testMaxQuantityLimit()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:45 COT" name="testMaxQuantityLimit" finished-at="2025-02-27T09:56:48 COT" duration-ms="2542" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[La cantidad del producto debería limitarse al máximo configurado (10) expected [10] but found [0]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: La cantidad del producto debería limitarse al máximo configurado (10) expected [10] but found [0]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.expresscart.tests.CartFunctionalityTest.testMaxQuantityLimit(CartFunctionalityTest.java:168)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMaxQuantityLimit -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:48 COT" name="tearDown" finished-at="2025-02-27T09:56:48 COT" duration-ms="523" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testMaxQuantityLimit status=FAILURE method=CartFunctionalityTest.testMaxQuantityLimit()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String,java.lang.String)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:48 COT" name="setUp" finished-at="2025-02-27T09:56:52 COT" duration-ms="3267" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[false]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testRemoveProductFromCart()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:56:52 COT" name="testRemoveProductFromCart" finished-at="2025-02-27T09:57:00 COT" duration-ms="8861" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[El carrito debería tener 2 productos expected [2] but found [0]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: El carrito debería tener 2 productos expected [2] but found [0]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.expresscart.tests.CartFunctionalityTest.testRemoveProductFromCart(CartFunctionalityTest.java:81)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testRemoveProductFromCart -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005]" started-at="2025-02-27T09:57:00 COT" name="tearDown" finished-at="2025-02-27T09:57:01 COT" duration-ms="346" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=testRemoveProductFromCart status=FAILURE method=CartFunctionalityTest.testRemoveProductFromCart()[pri:0, instance:com.expresscart.tests.CartFunctionalityTest@67d48005] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- com.expresscart.tests.CartFunctionalityTest -->
    </test> <!-- Shopping Cart Functionality Tests -->
  </suite> <!-- ExpressCart Shopping Cart Test Suite -->
</testng-results>
